### PR TITLE
Allow setting custom initialization vector for FileAccessEncrypted. Add export setting to set static seed for PCK encryption initialization vectors.

### PR DESCRIPTION
**Pull request by bruvzg**
- Allow setting custom initialization vector for `FileAccessEncrypted`.
- Add export setting to set static seed for PCK encryption initialization vectors, If set to non-zero value `seed + file hash` are used to generate IV, if zero IV is fully random (old behavior).

Fixes https://github.com/godotengine/godot/issues/98904

### CODE REVIEW

File: core/io/file_access.h [hunk 1]
```diff
@@ -109,6 +109,12 @@ class FileAccess : public RefCounted {
 
 	static FileCloseFailNotify close_fail_notify;
 
+#ifndef DISABLE_DEPRECATED
+	static Ref<FileAccess> _open_encrypted_compat_xxxxx(const String &p_path, ModeFlags p_mode_flags, const Vector<uint8_t> &p_key);
```
Comment (AThousandShips) [inline core/io/file_access.h:113]:
```suggestion
	static Ref<FileAccess> _open_encrypted_bind_compat_98918(const String &p_path, ModeFlags p_mode_flags, const Vector<uint8_t> &p_key);
```

File: misc/extension_api_validation/4.3-stable.expected [hunk 2]
```diff
@@ -108,3 +108,10 @@ GH-97257
 Validate extension JSON: Error: Field 'classes/EditorFeatureProfile/enums/Feature/values/FEATURE_MAX': value changed value in new API, from 8.0 to 9.
 
 New entry to the `EditorFeatureProfile.Feature` enum added. Those need to go before `FEATURE_MAX`, which will always cause a compatibility break.
+
+
+GH-xxxxx
```
Comment (akien-mga) [inline misc/extension_api_validation/4.3-stable.expected:113]:
```suggestion
GH-98918
```

**Review by akien-mga — APPROVED** [review]
Aside from the compat method name, seems good to me.

**Review by AThousandShips — APPROVED** [review]
Haven't tested but looks good

File: core/io/file_access_encrypted.cpp [hunk 3]
```diff
@@ -49,6 +49,15 @@ Error FileAccessEncrypted::open_and_parse(Ref<FileAccess> p_base, const Vector<u
 		writing = true;
 		file = p_base;
 		key = p_key;
+		if (p_iv.is_empty()) {
+			iv.resize(16);
+			for (int i = 0; i < 16; i++) {
+				iv.write[i] = Math::rand() % 256;
+			}
+		} else {
```
Comment (Bromeon) [inline core/io/file_access_encrypted.cpp:57]: I don't know the full context here, but does the init-vector not need to be _securely_ random? As repeated invocations of `Math::rand()` probably have rather low entropy...

Comment (bruvzg) [reply core/io/file_access_encrypted.cpp]: I do not think it matters in this case, but changed it to `CryptoCore::RandomGenerator` just in case.

**Pull request merged**